p2p/enode/nodedb.go

```diff
@@ -62,9 +62,10 @@ const (
 )
 
 const (
-	dbNodeExpiration = 24 * time.Hour // Time after which an unseen node should be dropped.
-	dbCleanupCycle   = time.Hour      // Time period for running the expiration task.
-	dbVersion        = 10
+	dbNodeExpiration     = 24 * time.Hour // Time after which an unseen node should be dropped.
+	dbCleanupCycle       = time.Hour      // Time period for running the expiration task.
+	dbVersion            = 10
+	dbSyncBytesThreshold = 20_000 // if we have about 20kb of dirty data , then flush to disk
```

Comment:
where `20kb` came from?

Reply:
From observing metrics, this seems to be the max. `TxDirty` number ever reached.

Reply:
1. What will happen if syncBytesThreshold reached? 
2. It must be reachable in usual scenarios or un-reachable?

Reply:
1. The data will be flushed to disk before the sync period has completed.

2. If we remove the sync period, then it will be reached and cause a flush in perpetuity. With the sync period (current configuration), it won't be reached under normal conditions, however if there is a burst of activity the syncBytesThreshold will be reached before the sync period completes.

Reply:
1. Inside .Commit, or in another thread?

Reply:
With SafeNoSync it's happening in a background thread.

Reply:
mdbx doesn't have any background threads. so, it can't do anything in background. 
it can do 2 things:
- App must create goroutine which will periodically call `mdbx_env_sync_ex` - `fsync` will happen here
- If burst of updates will happen and one of next thresholds will be reached: `syncperiod`/`syncbytes` - then `fsync` will happen inside next `.Commit`

Mentioned this 3 funcs in: https://github.com/erigontech/erigon/issues/13550

Our target is: greatly reduce probability of `fsync` inside `.Commit`